### PR TITLE
Carousel implementation

### PR DESCRIPTION
**Pull request by thecalcc**
NHUB-432

### CODE REVIEW

File: superdesk-common/package.json [hunk 1]
```diff
@@ -10,13 +10,16 @@
...
+    "watch": "tsc --watch",
     "build": "tsc && ts-node bundle-css.ts",
     "watch-css": "ts-node bundle-css.ts --watch",
     "test": "mocha",
     "debug-tests": "mocha --inspect-brk",
     "prepublish": "rm -rf dist && npm run build && npm run test"
   },
   "devDependencies": {
+    "@types/classnames": "^2.3.1",
+    "classnames": "^2.3.2",
```
Comment (tomaskikutis) [inline superdesk-common/package.json:22]: doesn't look used. If it was though - it'd be required during runtime - thus it should be in main dependencies - not dev dependencies

File: superdesk-common/package.json [hunk 2]
```diff
@@ -30,6 +33,8 @@
     "typescript": "^4.5.2"
   },
   "dependencies": {
+    "@types/node": "^20.9.0",
```
Comment (tomaskikutis) [inline superdesk-common/package.json:36]: types should go to dev dependencies

File: superdesk-common/src/ui/Carousel.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    componentDidMount(): void {
+        this.setState({
+            height: this.el?.clientHeight,
+        });
+    }
+
+    render() {
+        const productTemplate = (image: any) => {
```
Comment (tomaskikutis) [inline superdesk-common/src/ui/Carousel.tsx:30]: don't use `any` ;)

Comment (tomaskikutis) [inline superdesk-common/src/ui/Carousel.tsx:30]: why is it called product? Are you planning to sell it :D

File: superdesk-common/src/ui/thumb-carousel.css [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+
+.p-carousel-prev.p-disabled,
+.p-carousel-next.p-disabled {
+	visibility: hidden;
+	pointer-events: none;
+}
+
+.sd-thumb-carousel__btn--prev,
+.p-carousel-prev {
+	margin-right: -51px;
```
Comment (tomaskikutis) [inline superdesk-common/src/ui/thumb-carousel.css:261]: wouldn't `-100%` work? also go over hardcoded values and comment how did you come up with the value

**Comment by thecalcc** [comment]
> should this be closed as duplicate of #4386 ?

No, I reverted that one. You hadn't reviewed it so I had to revert it and open a new one. Please review the last two commits

**Review by tomaskikutis — APPROVED** [review]

**Pull request merged**